src/conventional-commits.ts

```diff
@@ -201,78 +180,62 @@ export class ConventionalCommits {
       this.headerPartial || preset.writerOpts.headerPartial;
     preset.writerOpts.mainTemplate =
       this.mainTemplate || preset.writerOpts.mainTemplate;
-
-    return new Promise((resolve, reject) => {
-      let content = '';
-      const stream = this.commitsReadable()
-        .pipe(conventionalCommitsParser(preset.parserOpts))
-        .pipe(new PostProcessCommits({objectMode: true}))
-        .pipe(conventionalChangelogWriter(context, preset.writerOpts));
-
-      stream.on('error', (err: Error) => {
-        return reject(err);
-      });
-
-      stream.on('data', (buffer: Buffer) => {
-        content += buffer.toString('utf8');
-      });
-
-      stream.on('end', () => {
-        return resolve(content.trim());
-      });
-    });
+    const parsedCommits = [];
+    for (const commit of this.commits) {
+      try {
+        const parsedCommit = postProcessCommits(
+          toConventionalChangelogFormat(parser(commit.message))
+        ) as CommitWithHash;
+        parsedCommit.hash = commit.sha;
+        parsedCommits.push(parsedCommit);
+      } catch (_err) {
+        // Commit is not in conventional commit format, it does not
```

Comment:
Actually, this makes it very easy to miss an important commit in the change log if it was not named properly. Maybe we could have a section in the change log listing unparsed commits as is?